sztp-agent/pkg/dhcp/bootstrap_url.go

```diff
@@ -0,0 +1,25 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+Copyright (C) 2022-2023 Intel Corporation
+Copyright (c) 2022 Dell Inc, or its subsidiaries.
+Copyright (C) 2022 Red Hat.
+*/
+
+// Package dhcp implements the DHCP client
+package dhcp
+
+import "log"
+
+// GetBootstrapURL returns the bootstrap URL
+func GetBootstrapURL(dhcpLeaseFile string) ([]string, error) {
+	url, err := getBootstrapURLViaLeaseFile(dhcpLeaseFile)
```

Comment:
you always try to parse lease file first here... why ?

1. if user gave me URL to run on, run on it, don't check dhcp leases or network manager
2. if user gave me leases file, parse it, don;t check network manager
3. if no URL and no LEASE file given by user, check network manager